src/Monitor/Monitor/ScheduledQueryRules/NewScheduledQueryRuleCommand.cs

```diff
@@ -0,0 +1,128 @@
+
+using Microsoft.Azure.Commands.Insights.OutputClasses;
+using Microsoft.Azure.Management.Monitor.Models;
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.Insights.ScheduledQueryRules
+{
+    /// <summary>
+    /// Create a ScheduledQueryRule Source object
+    /// </summary>
+    [Cmdlet(VerbsCommon.New, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ScheduledQueryRule", SupportsShouldProcess = true), OutputType(typeof(PSScheduledQueryRuleResource))]
+    public class NewScheduledQueryRuleCommand : ManagementCmdletBase
+    {
+
+        #region Cmdlet parameters
+
+        //
+        // Summary:
+        //     Gets or sets source (Query, DataSourceId, etc.) for rule.
+        [Parameter(Mandatory = true, HelpMessage = "The scheduled query rule source")]
+        [ValidateNotNullOrEmpty]
+        public PSScheduledQueryRuleSource Source { get; set; }
+
+        //
+        // Summary:
+        //     Gets or sets schedule (Frequnecy, Time Window) for rule.
+        [Parameter(Mandatory = true, HelpMessage = "The scheduled query rule schedule")]
+        [ValidateNotNullOrEmpty]
+        public PSScheduledQueryRuleSchedule Schedule { get; set; }
+
+        //
+        // Summary:
+        //     Gets or sets action needs to be taken on rule execution.
+        [Parameter(Mandatory = true, HelpMessage = "The scheduled query rule Alerting Action")]
+        [ValidateNotNullOrEmpty]
+        public PSScheduledQueryRuleAlertingAction Action { get; set; }
+
+        //
+        // Summary:
+        //     Region where alert is to be created
+        [Parameter(Mandatory = true, HelpMessage = "The location for this alert")]
+        [ValidateNotNullOrEmpty]
+        public string Location { get; set; }
+
+        //
+        // Summary:
+        //     Alert description
+        [Parameter(Mandatory = false, HelpMessage = "The description for this alert")]
+        public string Description { get; set; }
+
+        //
+        // Summary:
+        //     Alert name
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The alert name")]
+        [ResourceNameCompleter("Microsoft.insights/scheduledqueryrules", nameof(ResourceGroupName))]
+        public string RuleName { get; set; }
+
+        //
+        // Summary:
+        //     Resource tags
+        [Parameter(Mandatory = false, HelpMessage = "Resource tags")]
+        public IDictionary<string, string> Tag;
+
+        //
+        // Summary:
+        //     Alert status - enabled or not
+        [Parameter(Mandatory = true, HelpMessage = "The azure alert state - valid values - true, false")]
+        [ValidateSet("true", "false")]
+        [PSArgumentCompleter("true", "false")]
+        public string Enabled { get; set; }
```

Comment:
@shpimpal this shouldn't be a `string` -- you should use a `SwitchParameter` to represent this parameter. What is the default value for this resource? If it's enabled, then we should change this to a parameter called `Disable`, and vice versa if the default behavior is disabled.

Reply:
It's directly passed as a string to the parent resource class. It is a mandatory parameter so there is no default value.

Reply:
I tried changing it to bool, the cmdlet accepts only 0 or 1 (or any number) as a value. If I provide true or false, it takes it as a string and fails as the argument is of type bool.